api/krusty/fnplugin_test.go

```diff
@@ -504,313 +504,94 @@ func skipIfNoDocker(t *testing.T) {
 }
 
 func TestFnContainerGenerator(t *testing.T) {
-	t.Skip("wait for #3881")
 	skipIfNoDocker(t)
-
-	// Function plugins should not need the env setup done by MakeEnhancedHarness
 	th := kusttest_test.MakeHarness(t)
-
-	th.WriteK(".", `
```

Comment:
Is there a reason not to use these test helpers? I think the following should work:

```diff
diff --git a/api/krusty/fnplugin_test.go b/api/krusty/fnplugin_test.go
index 8d3e0e196..1953adf72 100644
--- a/api/krusty/fnplugin_test.go
+++ b/api/krusty/fnplugin_test.go
@@ -507,15 +507,14 @@ func TestFnContainerGenerator(t *testing.T) {
 	skipIfNoDocker(t)
 	th := kusttest_test.MakeHarness(t)
 	o := th.MakeOptionsPluginsEnabled()
-	fSys := filesys.MakeFsOnDisk()
-	b := MakeKustomizer(&o)
 	tmpDir, err := filesys.NewTmpConfirmedDir()
 	assert.NoError(t, err)
-	assert.NoError(t, fSys.WriteFile(filepath.Join(tmpDir.String(), "kustomization.yaml"), []byte(`
+
+	th.WriteK(tmpDir.String(), `
 generators:
 - project-service-set.yaml
-`)))
-	assert.NoError(t, fSys.WriteFile(filepath.Join(tmpDir.String(), "project-service-set.yaml"), []byte(`
+`)
+	th.WriteF(filepath.Join(tmpDir.String(), "project-service-set.yaml"), `
 apiVersion: blueprints.cloud.google.com/v1alpha1
 kind: ProjectServiceSet
 metadata:
@@ -528,9 +527,8 @@ spec:
   services:
     - compute.googleapis.com
   projectID: foo
-`)))
-	m, err := b.Run(fSys, tmpDir.String())
-	assert.NoError(t, err)
+`)
+	m := th.Run(tmpDir.String(), o)
 	actual, err := m.AsYaml()
 	assert.NoError(t, err)
 	assert.Equal(t, `apiVersion: serviceusage.cnrm.cloud.google.com/v1beta1
```

Reply:
It does make sense. I followed your suggestion.